packages/embark/src/lib/modules/blockchain_connector/provider.js

```diff
@@ -18,8 +18,7 @@ class Provider {
     this.nonceCache = {};
 
     this.events.setCommandHandler("blockchain:provider:contract:accounts:get", cb => {
-      const accounts = this.accounts.map(a => a.address);
```

Comment:
Does someone know why it was `a.address`?

Reply:
Wouldn't be surprised if that was related to a (breaking) change between web3 js releases

Reply:
@alaibe , I think it's because this function should only return addresses and `this.accounts` is supposed to contain actual web3 account objects (which have `.address`).

So you probably need to change this to an `if` that checks if it's an object or just an address.

The reason we only get an address is probably since Andre changed the provider to only use the web3 accounts for deployment only and if the blockchain accounts have nothing, they are just an address.

Reply:
Updated, let me know if ok now

Reply:
Just saw that Eric did the same change: https://github.com/embark-framework/embark/pull/1368/files